Create first version of documentation

jksf opened the pull request.

Commit:
Initial commit

Commit:
Create first version of documentation

jksf commented:
@biern I managed to generate the documentation using `doxity`. It was very tricky, and their official guide is broken. Currently it requires global `truffle` and `solc` (not `solcjs`) installed. I generated the HTML assets this way.
In a next PR, I will prepare `docker-compose` to avoid these dependencies.

jksf commented:
@biern Can you take a look at this? I'd like to move the documentation work forward.

biern commented:
Sure. Will do tomorrow.

2018-01-06 22:07 GMT+01:00 Jakub Stefanski <notifications@github.com>:

> @biern <https://github.com/biern> Can you take a look at this? I'd like
> to move the documentation work forward.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/SignHash/signhash-contracts-docs/pull/2#issuecomment-355776869>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AACjhkcd6Lo8Nlk6iZF65Vvm2jofNkOoks5tH-B3gaJpZM4RBTwY>
> .
>



-- 
Pozdrawiam
  Marcin Biernat

jksf commented:
@biern Thanks, it's not urgent but it stuck here for a while :-)

biern commented:
So I'm not sure how to run it, I would appreciate an instruction in the readme.
- I tried serving files from `docs` but the app generates invalid paths
`http://localhost:8000/signhash-contracts-docs/docs/AddressSet/` (invalid) vs `http://localhost:8000/docs/docs/AddressSet/` (valid).
- If I navigate to contract docs manually there's still something wrong, Clicking on the tabs doesn't do anything.
- What's the motiviation for keeping docs outside of the contracts repo?

Commit:
Build docs inside docker

Doxity project is a mess. It requires several global dependencies.
This commit makes it sane using docker.

Commit:
Update README with build instructions

Commit:
Fix docs generation and link prefixes

Commit:
Update docs
